### ISSUE TITLE
shinden.pl [Timer & Anti-Adblock]

### ISSUE THREAD

**Issue by kulfoon** [uBlockOrigin/uAssets#5551]
### URL(s) where the issue occurs

Example url: `https://shinden.pl/episode/10831-steins-gate/view/87288#column-one-inner`

### Describe the issue

<details not-open>
<summary>READ THIS FIRST!</summary>

- ignore the other issues (Main Anti-Adblock + Click Popup ==> https://github.com/MajkiIT/polish-ads-filter/issues/12976), as they're already fixed and the fixes will be soon added to regional polish filter lists, so no need to fix them nor add to uAssets, so just avoid the noise and temporarily add these solutions to My Filters:
```
@@||criteo.net/js/ld/publishertag.js$script,third-party,domain=shinden.pl
@@/^https:\/\/www\.[a-z]{7,14}\.com\/[a-z]{1,4}\.js/$script,third-party,domain=shinden.pl
shinden.pl##+js(popads-dummy.js)
```

As for the regex filter it might be changing so always check the newest  version here:
https://github.com/MajkiIT/polish-ads-filter/issues/12976#issuecomment-489317031

- ignore "403 Forbidden (467)" error on a blank white page (https://github.com/MajkiIT/polish-ads-filter/issues/12976#issuecomment-490037603), which might sometimes
 appears especially while visiting the website for first time, just reload the page and it should load properly the second time.

- if solution/filter will be found, we can/prefer to add it to polish regional lists to avoid cluttering uAssets with regional filters, if no solution will be found then we can deal with 5 sec timer as well or perhaps one of Big Men will create a solution.
</details>

Now the current reported issues are:

Issue 1: 5 sec timer when trying to play video

Issue 2 (related): show video button replaced by anti-adblock button when trying to bypass the issue 1 (the timer) by adding one of the following filters:

```
shinden.pl##+js(nano-setTimeout-booster.js, value_count)
shinden.pl##+js(nano-setTimeout-booster.js)
shinden.pl##+js(setTimeout-logger.js)
shinden.pl##+js(setTimeout-defuser.js)

or other example filters:

shinden.pl##+js(setInterval-logger.js)
shinden.pl##+js(setInterval-defuser.js)
shinden.pl##+js(nano-setInterval-booster.js)
```
After clicking the show/play button it's redirecting to:
`https://forum.shinden.pl/index.php?threads/abp-jak-odblokowa%C4%87-problem-z-odtwarzaczami.22678/` which says:
> You get here, probably because you do not see the player's load button, and the message How to Unblock.
> This is due to the fact that our script has detected forbidden activities: blocking ads. (Yes, advertising is one of the forms of maintaining shinden.pl)
> If, despite the ad block off, the message appears to you, this is one of the options:...

So shinden.pl has a script that detects/defuses setTimeout / setInterval modifications or perhaps it even directly defuses setTimeout & setInterval related uBO scriplets.

Also notice that the whole website is agressive: it's obfuscated and generates requests to random domains with random script names, even people complain about it's aggressiveness on the fourm: `https://forum.shinden.pl/index.php?threads/abp-jak-odblokowa%C4%87-problem-z-odtwarzaczami.22678/`

### Screenshot(s)

**1 Before bypassing the timer:**

![SS_012](https://user-images.githubusercontent.com/43844449/57293996-92308680-70c6-11e9-88c2-ac965e2f0437.png)
("Pokaż" = "Show" , which simply means "show video")

![ANIM_638](https://user-images.githubusercontent.com/43844449/57293876-3cf47500-70c6-11e9-9bd9-06375071e0a6.gif)

**2 After trying to bypass the timer:**

![SS_011](https://user-images.githubusercontent.com/43844449/57294002-96f53a80-70c6-11e9-80c8-6f74a433fdbf.png)
("Jak odblokować" = "How to unblock")

![ANIM_911](https://user-images.githubusercontent.com/43844449/57293919-5c8b9d80-70c6-11e9-930a-6592837ed246.gif)

### Versions

- Browser/version: Firefox 66.0.4 64bit / Opera 60 64bit
- uBlock Origin version:  1.18.16

### Settings

Default PL 
(regional lists:
POL: Oficjalne Polskie Filtry do AdBlocka, uBlocka Origin i AdGuarda​​​​​
POL: Oficjalne polskie filtry przeciwko alertom o Adblocku
and Polish Privacy Filters​ (`https://raw.githubusercontent.com/olegwukr/polish-privacy-filters/master/adblock.txt`)

### Notes

**Comment by okiehsch** [comment]
I always get a 404 error.

![image](https://user-images.githubusercontent.com/16838044/57325744-0b51cd00-710b-11e9-87ee-2fd71564313b.png)

**Comment by krystian3w** [comment]
Server side timer... https://github.com/NanoAdblocker/NanoFilters/issues/41#issuecomment-388198988

**Comment by kulfoon** [comment]
Before reporting the issue here, I searched both polish and uassets repos for the timer issue, but completely forgot to search reek's or penguin's repos, now I've also found 2 more timer related reports as well: 
https://github.com/reek/anti-adblock-killer/issues/2955
https://github.com/reek/anti-adblock-killer/issues/3266

In the second one, have found another - a year older (June 2017) penguin's comment: https://github.com/reek/anti-adblock-killer/issues/3266#issuecomment-306048821 saying the same what he said in his comment krystian3w attached above (May 2018).

Another year passed and now we have May 2019, if the timer issue still didn't changed and the timer is validated a server side, then the issue can be closed.

**Comment by krystian3w** [comment]
If possible please try run site on proxy / vpn maybe possible easy create scriplet better as regex network exception:

>  random domains with random script names, even people complain about it's aggressiveness on the fourm

```regex
@@||criteo.net/js/ld/publishertag.js$script,third-party,domain=shinden.pl
@@/^htt(p|ps):\/\/www\.[a-z]{7,14}\.com\/[a-zA-Z]{2,4}\.js/$script,third-party,domain=shinden.pl
```

```
https://www.rfmbvlaphdud.com/nn.js
https://www.ffpnlrnmfyeqx.com/awj.js
https://www.lpbyadpvf.com/up.js
https://www.vuvcwrxn.com/rbzq.js
https://www.bexbpzultczaa.com/wi.js

https://www.oezgivtasc.com/zrx.js
https://www.vvqpavyfkr.com/ierb.js

https://www.ogqewglysfc.com/el.js
```

**Comment by jspenguin2017** [comment]
I was able to reproduce anti-adblock, see: https://github.com/NanoMeow/QuickReports/issues/415

Haven't got time to find a fix yet.

**Comment by kulfoon** [comment]
To clarify, the issue I reported is not about the Main Anti-Adblock & Click-Popup, but about Timer Anti Adblock - see the spoiler in my first comment https://github.com/uBlockOrigin/uAssets/issues/5551#issue-441272289 "READ THIS FIRST!" unless the both Anti-Adblocks issues are somehow related and triggered by the same code...

As for the Main Anti-Adblock issue which is discussed in other threads: 
https://github.com/MajkiIT/polish-ads-filter/issues/12976
https://github.com/NanoMeow/QuickReports/issues/415 
the solution has been already created - see https://github.com/MajkiIT/polish-ads-filter/issues/12976#issuecomment-489317031, but not yet added to polish filters lists.

Perhaps this guy https://github.com/NanoMeow/QuickReports/issues/415#issuecomment-489424693 is not aware of the solution...

As for this guy https://github.com/uBlockOrigin/uAssets/issues/5551#issuecomment-490241754 he didn't say before that a scriplet solution is required, as polish filters lists support ABP, thus stick to ABP compatible (whitelisting) solutions rather than to uBO only compatible scriplet injections, as mentioned here: https://github.com/MajkiIT/polish-ads-filter/issues/8544#issuecomment-411511204 + https://github.com/MajkiIT/polish-ads-filter/issues/8544#issuecomment-411540732. He suddenly says it just right now...now me being aware of a need a for a scriplet solution I'll try to find one, or perhaps penguin will find one meantime.

**Comment by jspenguin2017** [comment]
Ah, I was too tired last night. Ignore my comment above...

**Comment by kulfoon** [comment]
> kulfoon : https://github.com/uBlockOrigin/uAssets/issues/5551#issuecomment-490239585 : (...) if the timer issue still didn't changed and the timer is validated a server side, then the issue can be closed.

jspenguin2017 - so am I right that I can close it here as unfixable by pure uBO and move/transfer it (paste the link) to the main issue: https://github.com/NanoMeow/QuickReports/issues/415

**Comment by jspenguin2017** [comment]
Well, if someone wants to challenge it, I'm not going to stop them. But I'm not spending any time on trying to get past the 5 seconds timer.

**Comment by krystian3w** [comment]
Latin566 created scriplets: https://github.com/MajkiIT/polish-ads-filter/issues/12976#issuecomment-491094068

> Can someone test it?
> ```
> shinden.pl##+js(abort-on-property-write.js, _pop)
> shinden.pl##+js(noeval-if.js, Criteo)
> ```

**Comment by garggul** [comment]
> krystian3w : 
> 
> Latin566 created scriplets: [MajkiIT/polish-ads-filter#12976 (comment)](https://github.com/MajkiIT/polish-ads-filter/issues/12976#issuecomment-491094068)
> 
> > Can someone test it?
> > ```
> > shinden.pl##+js(abort-on-property-write.js, _pop)
> > shinden.pl##+js(noeval-if.js, Criteo)
> > ```

https://github.com/uBlockOrigin/uAssets/issues/5551#issuecomment-490392512 Filters you posted are for the the Main-Anti-Adblock issue, and not for the Timer-Anti-Adblock issue, which is the case here, however let this thread starting from your comment be also the Main-Anti-Adblock issue (perhaps you thinked the same to avoid creating unnecessarily a new issue and splitting the whole website into dwo different issues), but hey - the scriplets most probably will be added to regional polish filters lists anyway, so no need to duplicate them, however they can test them to find if the filters don't break the website's functionality (perhaps that was the case you posted it here).
